### PR TITLE
LIReC gRPC server workaround

### PR DESCRIPTION
**Pull request by krachwal**
Created a gRPC wrapper for LIReC as a workaround to address the compatibility issue when running LIReC in the same venv as ResearchTools, both libraries being needed by the web portal.

It wraps a single function call in LIReC that takes in the long decimal number ResearchTools computes and returns likely closed forms or pretty formulas (i.e. turns the long number into the formula in blue):

image

Note: This is part of a larger changeset. More PRs to follow.

### CODE REVIEW

File: python-backend/requirements.txt [hunk 1]
```diff
@@ -8,4 +8,3 @@ requests==2.31.0
 sympy==1.12
 websockets==12.0
 ramanujan@https://github.com/RamanujanMachine/ResearchTools/archive/refs/heads/master.tar.gz
-LIReC@git+https://github.com/RamanujanMachine/LIReC.git@main
```
Comment (krachwal) [inline python-backend/requirements.txt:11]: This is now an indirect dependency

File: .github/workflows/cloud-deploy.yml [hunk 2]
```diff
@@ -1,17 +1,15 @@
 name: Build and Deploy to Cloud
 
 on:
-  push:
-#    branches:
-#      - main
```
Comment (krachwal) [inline .github/workflows/cloud-deploy.yml:6]: changing the trigger to manual dispatch so it stops deploying with every commit

File: Dockerfile [hunk 3]
```diff
@@ -5,18 +5,19 @@ LABEL description="Ramanujan Machine Web Portal"
...
-# install node
-RUN apt-get update && apt-get -y install git python3 python3-pip python3.10-venv ca-certificates curl gnupg libpq-dev libgmp-dev libmpfr-dev libmpc-dev
-
-RUN mkdir -p /etc/apt/keyrings
-RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
-ENV NODE_MAJOR=20
-RUN echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
-
-RUN apt-get update && apt-get -y install nodejs
+# install system dependencies
```
Comment (krachwal) [inline Dockerfile:13]: updating the node installation procedure

File: Dockerfile [hunk 4]
```diff
@@ -35,8 +36,8 @@ COPY ./python-backend/requirements.txt ./
 RUN python3 -m venv venv
 RUN chmod +x venv/bin/activate
 RUN ./venv/bin/activate
-RUN pip3 install -r requirements.txt
-RUN pip3 install uvicorn
+RUN pip3 install -r requirements.txt --break-system-packages
+RUN pip3 install uvicorn --break-system-packages
```
Comment (krachwal) [inline Dockerfile:41]: suddenly necessary to add `--break-system-packages` even though running in venv

File: README.md [hunk 5]
```diff
@@ -3,12 +3,15 @@
 ## Run with Docker
 To run the web portal using the Dockerfile:
```
Comment (krachwal) [inline README.md:5]: neglected to include that a Wolfram app id is necessary for the web portal to run

**Review by davidbrownell — APPROVED** [review]

**Pull request merged**